core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -343,27 +362,31 @@
                                      (.polyReader [:union [:null t/temporal-col-type]]))
               valid-to-rdr (some-> (.vectorForName in-rel "xt$valid_to")
                                    (.polyReader [:union [:null t/temporal-col-type]]))]
-          (dotimes [idx row-count]
-            (let [row-id (.nextRowId live-chunk)]
-              (.writeRowId live-table row-id)
-              (.copyRow table-copier idx)
-
-              (let [eid (t/get-object (.getVector id-col) (.getIndex id-col idx))
-                    new-entity? (not (.isKnownId iid-mgr table eid))
-                    iid (.getOrCreateInternalId iid-mgr table eid row-id)
-                    valid-from (if (and valid-from-rdr (= 1 (.read valid-from-rdr idx)))
-                                 (.readLong valid-from-rdr)
-                                 current-time-µs)
-                    valid-to (if (and valid-to-rdr (= 1 (.read valid-to-rdr idx)))
-                               (.readLong valid-to-rdr)
-                               util/end-of-time-μs)]
-                (when (> valid-from valid-to)
-                  (throw (err/runtime-err :xtdb.indexer/invalid-valid-times
-                                          {:valid-from (util/micros->instant valid-from)
-                                           :valid-to (util/micros->instant valid-to)})))
-
-                (.logPut log-op-idxer iid row-id valid-from valid-to)
-                (.indexPut temporal-idxer iid row-id valid-from valid-to new-entity?)))))))))
+          ;; TODO revisit to check if copying to content log can be done more efficiently (see also sql-update)
+          (with-open [^ValueVector content-vec (vw/irel->struct-vec allocator content-rel)]
```

Comment:
@jarohen This is kind of hack. In the content-log I would like to have a vec which I get in `put` but not here. Open for suggestions, but we can also revisit once things stablilize.

Reply:
I'd lean more on the `IRowCopier` interface here rather than a separate `.logPut` on the content-log - live-chunk is probably more of an inspiration than the existing log indexer for this one. Let's pair :slightly_smiling_face:

Reply:
See main PR comment.